File: 283-move-zeroes/move-zeroes.c
void moveZeroes(int* nums, int numsSize) {
    int low =0;

    int high =1;

    while(high<numsSize && low <numsSize){
        if((nums[low] == 0) && (nums[high] != 0)){
            int temp =nums[high];
            nums[high]= nums[low];
            nums[low]=temp;
            low++;
            high++;
        }else if(nums[low] != 0) {
            low++;
            if(high<=low){
                high=low+1;
            }
        }
        else high++;
    }

}

File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        j = 0
        for i in range(0 , len(nums)):
            if(nums[i] != 0 ):
                temp = nums[i]
                nums[i] = 0
                nums[j] = temp
                j = j+1
        